### PR TITLE
Update getChildrenSummaryTree to handle Diagnosticable as input.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

**Review by jacob314 — COMMENTED** [review]
Needs a test verifying getChildrenSummaryTree can be called with a Diagnosticable.

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 1]
```diff
@@ -1758,9 +1758,24 @@ mixin WidgetInspectorService {
   }
 
   List<Object> _getChildrenSummaryTree(String? diagnosticsNodeId, String groupName) {
-    final DiagnosticsNode? node = toObject(diagnosticsNodeId) as DiagnosticsNode?;
+    final Object? theObject = toObject(diagnosticsNodeId);
     final InspectorSerializationDelegate delegate = InspectorSerializationDelegate(groupName: groupName, summaryTree: true, service: this);
-    return _nodesToJson(node == null ? const <DiagnosticsNode>[] : _getChildrenFiltered(node, delegate), delegate, parent: node);
+
+    // TODO(polina-c): remove this, when DevTools stops sending DiagnosticsNode.
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:1764]: nit: file a bug on DevTools for this and mention it here to make it easier to track.

Comment (polina-c) [reply packages/flutter/lib/src/widgets/widget_inspector.dart:1764]: This is tracked as subtask of https://github.com/flutter/devtools/issues/3951. added to comment.

File: packages/flutter/test/widgets/widget_inspector_test.dart [hunk 2]
```diff
@@ -2397,6 +2398,115 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
       expect(childJson['chidlren'], isNull);
     }, skip: !WidgetInspectorService.instance.isWidgetCreationTracked()); // [intended] Test requires --track-widget-creation flag.
 
+    testWidgets('ext.flutter.inspector.getRootWidgetSummaryTree, Dianosticable', (WidgetTester tester) async {
```
Comment (jacob314) [inline packages/flutter/test/widgets/widget_inspector_test.dart:2402]: nit: typo on Diagnosticable. Also, prefer making test case names sentence fragments so
perhaps
`'ext.flutter.inspector.getRootWidgetSummaryTree on Diagnosticable'`

**Review by jacob314 — APPROVED** [review]
lgtm
once the two minor nits are resolved.

**Pull request merged**